File: LeetCode/2492.cpp
class Solution {
public:
    vector<pair<int, int>> G[100000];
    int mp = 1e9;
    bool vis[100000];
    void dfs(int v){
        if(vis[v]) return;
        vis[v] = 1;
        for(pair<int, int> e : G[v]){
            mp = min(mp, e.second);
            dfs(e.first);
        }
    }
    int minScore(int n, vector<vector<int>>& roads) {
        for(vector<int> &v : roads){
            G[v[0]-1].push_back({v[1]-1, v[2]});
            G[v[1]-1].push_back({v[0]-1, v[2]});
        }
        dfs(0);
        return mp;
    }
};

File: LeetCode/2492.py
class Solution:
    def minScore(self, n: int, roads: List[List[int]]) -> int:
        adj = [[] for _ in range(n)]
        for a, b, d in roads:
            adj[a-1].append((b-1, d))
            adj[b-1].append((a-1, d))

        vis = [False] * n
        vis[0] = True
        stack = [0]
        min_edge = 1e67
        while stack:
            v = stack.pop()
            for u, d in adj[v]:
                min_edge = min(min_edge, d)
                if vis[u]:
                    continue
                vis[u] = True
                stack.append(u)
        
        return min_edge